contracts/Core/BlockManager.sol

```diff
@@ -173,27 +174,38 @@ contract BlockManager is Initializable, ACL, BlockStorage {
         }
     }
 
-    function confirmBlock() external initialized onlyRole(parameters.getBlockConfirmerHash()) {
+    function claimBlockReward() public initialized checkState(parameters.confirm()){
+        uint256 epoch = parameters.getEpoch();
+        uint256 stakerId = stakeManager.getStakerId(msg.sender);
+        require(stakerId > 0, "Structs.Staker does not exist");
+        require(blocks[epoch].proposerId == 0, "Block already confirmed");
+
+        for (uint8 i = 0; i < proposedBlocks[epoch].length; i++) {
+            if (proposedBlocks[epoch][i].valid) {
+                require(proposedBlocks[epoch][i].proposerId == stakerId, "Block can be confirmed by proposer of the block");
```

Comment:
if first block isnt proposed by staker, the entire tx will fail?

Reply:
The first valid block, not the first block itself. The staker that has proposed the first valid block can only confirm the block